nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -199,6 +208,24 @@ def main():
         if os.readlink(system_dir(*gen)) == args.default_config:
             write_loader_conf(*gen)
 
+    memtest_entry_file = "@efiSysMountPoint@/loader/entries/memtest86.conf"
+    if os.path.exists(memtest_entry_file):
+        os.unlink(memtest_entry_file)
+    shutil.rmtree("@efiSysMountPoint@/efi/memtest86", ignore_errors=True)
+    if "@memtest86@" != "":
+        mkdir_p("@efiSysMountPoint@/efi/memtest86")
+        for path in glob.iglob("@memtest86@/*"):
+            if os.path.isdir(path):
+                shutil.copytree(path, os.path.join("@efiSysMountPoint@/efi/memtest86", os.path.basename(path)))
+            else:
+                shutil.copy(path, "@efiSysMountPoint@/efi/memtest86/")
+
+        memtest_entry_file = "@efiSysMountPoint@/loader/entries/memtest86.conf"
+        memtest_entry_file_tmp_path = "%s.tmp" % memtest_entry_file
+        with open(memtest_entry_file_tmp_path, 'w') as f:
+            f.write(MEMTEST_BOOT_ENTRY)
+        os.rename(memtest_entry_file_tmp_path, memtest_entry_file)
```

Comment:
The above code does the following:

1. Delete `/boot/loader/entries/memtest86.conf` if it exists.
2. Delete the `/boot/efi/memtest86/` directory.  (This is the directory that contains the actual memtest86 UEFI app, called `BOOTX64.efi`.)
3. If `boot.loader.systemd-boot.memtest86.enable` is set to `true`, the `@memtest86@` string will be replaced with the path to the `memtest86-efi` derivation, so the following things will be done.  (If `boot.loader.systemd-boot.memtest86.enable` is set to `false`, none of the following will be done.)
    1. Create the `/boot/efi/memtest86/` directory.
    2. Copy all the files from the `memtest86-efi` derivation to the `/boot/efi/memtest86/`.
    3. Create the `/boot/loader/entries/memtest86.conf` file.

The above is relatively similar to what this script is doing for the actual nixos generation loader files.